hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/Namesystem.java

```diff
@@ -63,4 +63,16 @@
    * directories. Create them if not.
    */
   void checkAndProvisionSnapshotTrashRoots();
+
+  /**
+   * Release read lock with operation name.
+   * @param opName
+   */
+  void readUnlock(String opName);
+
+  /**
+   * Release write lock with operation name.
+   * @param opName
+   */
+  void writeUnlock(String opName);
```

Comment:
How about moving the new methods to RwLock since it has all lock-related methods? Although RwLock is not `@InterfaceAudience.Private`, I think we can add new methods there if the target version is 3.4.0.

Reply:
Thanks @tasanuma for your review and comment. I agree with you.